### ISSUE TITLE
OpenBVE always returns to main screen just after new game data loaded

### ISSUE THREAD

**Issue by Symbian9** [leezer3/OpenBVE#401]
OpenBVE: `1.7.0.1`

  - https://vps.bvecornwall.co.uk/OpenBVE/Stable/OpenBVE-1.7.0.1.deb
  - https://vps.bvecornwall.co.uk/OpenBVE/Stable/OpenBVE-1.7.0.1.zip

### Description

*OpenBVE* returns to main screen just after new game data loaded.

### Reproduction

1. Launch *OpenBVE*;
2. Install @MarcRiera's `FCMB_L2_v1.0.obp` and `FCMB_2000_v1.3.obp` addons;
3. Start new game;

Here is new game configuration screen:

![pic.1](https://i.imgur.com/lotNXDd.png)

Here is new game loading screen (I uncheck default train because already set *FCMB 2000* manually):

![pic.2](https://i.imgur.com/kPUgwtv.png)

And here is main screen shown just after new game data loaded:

![pic.3](https://i.imgur.com/jDTYdv9.png)

When I click on <kbd>Close</kbd> button window closed, but in process manager I see that *OpenBVE* process still running, so I also terminated it manually for prevent CPU overloading.

#### Route

- https://github.com/MarcRiera/FCMB-L2/releases/download/v1.0/FCMB_L2_v1.0.obp

#### Train

- https://github.com/MarcRiera/FCMB-2000/releases/download/v1.3/FCMB_2000_v1.3.obp

#### Logs

I launch it throw Terminal and here its log:

```
$ openbve
Cannot connect to server socket err = No such file or directory
Cannot connect to server request channel
jack server is not running or cannot be started
JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for 4294967295, skipping unlock
JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for 4294967295, skipping unlock
Terminated
```

> **NOTE:** `Terminated` notification shown here because I manully kill process because it not stopped after closing main screen.

### Related information 

- Operating system: *Debian 9.x (x86_64)*
- Method of control: *Keyboard*

**Comment by leezer3** [comment]
I cannot reproduce this on a fresh virtual machine.

**However:**
Your terminal log somewhat suggests to me that your audio server (jack) isn't working right.
The sim *ought* to prompt if it fails to open the audio device, but it's possible that this isn't happening.

Any chance of seeing the most recent crash log (if it's generated one)- That can be found by hitting the **Report Problem** link in the bottom left of the main screen :)

**Comment by leezer3** [comment]
http://bveworldwide.forumotion.com/t1791-using-a-controller-with-openbve#top

Appears to be an identical problem :(
Also suggests that it's something changed in the 1.7 series which is the issue. Need a log from someone....

**Comment by Symbian9** [comment]
Here are logs:

### Previous log

<details>

```
openBVE Log: 10/12/2019 2:07:44 PM

14:07:45  Initialising game window of size 960 x 600
14:07:46  Game window initialised successfully.
14:07:47  Loading route file: /home/mint/.config/OpenBve/LegacyContent/Railway/Route/Metro de Barcelona - L3/2000 Zona Universitària.csv
14:07:47  INFO: Route file hash 3B878F9F7AA5621BB8534483F98AAAF890A35898EE52989C8145CEC88D5077C0
14:07:47  /home/mint/.config/OpenBve/LegacyContent/Railway : Railway folder found.
14:07:47  Route file format is: CSV
14:08:58  Route file loaded successfully.
14:08:59  Loading AI train: /home/mint/.config/OpenBve/LegacyContent/Train/FCMB 2000
14:09:00  Loading sound.cfg file: /home/mint/.config/OpenBve/LegacyContent/Train/FCMB 2000/Sound.cfg
14:09:20  Loading player train: /home/mint/.config/OpenBve/LegacyContent/Train/FCMB 2000
14:09:20  Loading sound.cfg file: /home/mint/.config/OpenBve/LegacyContent/Train/FCMB 2000/Sound.cfg
14:09:20  Loading train panel: /home/mint/.config/OpenBve/LegacyContent/Train/FCMB 2000/Panel.animated
14:09:23  Train panel loaded sucessfully.
14:09:23  Loading train plugin: /home/mint/.config/OpenBve/LegacyContent/Train/FCMB 2000/FCMB.dll
14:09:24  Train plugin loaded successfully.
14:09:29  The route and train loader encountered the following critical error: Failure has occurred while loading a type.
14:09:29  WARNING: Program crashing. Creating CrashLog file: /home/mint/.config/OpenBve/Settings/OpenBVE Crash- 2019.10.12[14.09].log
14:09:44  WARNING: Unrecognised key #restartarguments detected in filesystem.cfg
```

</details>

### Previous crash log

<details>

```
10/12/2019 2:09:29 PM
OpenBVE 1.7.0.1 Crash Log
Program is running on the Linux backend
Current screen resolution is: Windowed 960px X 600px 
Current routefile is: /home/mint/.config/OpenBve/LegacyContent/Railway/Route/Metro de Barcelona - L3/2000 Zona Universitària.csv
Current train is: /home/mint/.config/OpenBve/LegacyContent/Train/FCMB 2000
Current train plugin is: FCMB.dll
The current routefile caused the following exception: 
System.TypeLoadException: Failure has occurred while loading a type.
  at OpenBve.TrainManager+Car.UpdateCarSectionElement (System.Int32 SectionIndex, System.Int32 GroupIndex, System.Int32 ElementIndex, OpenBveApi.Math.Vector3 Position, OpenBveApi.Math.Vector3 Direction, OpenBveApi.Math.Vector3 Up, OpenBveApi.Math.Vector3 Side, System.Boolean Show, System.Double TimeElapsed, System.Boolean ForceUpdate, System.Boolean EnableDamping) [0x001cf] in <eb4ae41797e5478ba3982f0076e9d8ed>:0 
  at OpenBve.TrainManager+Car.UpdateObjects (System.Double TimeElapsed, System.Boolean ForceUpdate, System.Boolean EnableDamping) [0x00299] in <eb4ae41797e5478ba3982f0076e9d8ed>:0 
  at OpenBve.TrainManager+Car.ChangeCarSection (OpenBve.TrainManager+CarSectionType newCarSection) [0x00144] in <eb4ae41797e5478ba3982f0076e9d8ed>:0 
  at OpenBve.TrainManager+Train.Update (System.Double TimeElapsed) [0x0015a] in <eb4ae41797e5478ba3982f0076e9d8ed>:0 
  at OpenBve.TrainManager+Train.Initialize () [0x00026] in <eb4ae41797e5478ba3982f0076e9d8ed>:0 
  at OpenBve.OpenBVEGame.SetupSimulation () [0x00512] in <eb4ae41797e5478ba3982f0076e9d8ed>:0 
  at OpenBve.OpenBVEGame.LoadingScreenLoop () [0x00171] in <eb4ae41797e5478ba3982f0076e9d8ed>:0 
  at OpenBve.OpenBVEGame.OnLoad (System.EventArgs e) [0x000b3] in <eb4ae41797e5478ba3982f0076e9d8ed>:0 
  at OpenTK.GameWindow.OnLoadInternal (System.EventArgs e) [0x00000] in <6764f93f9a3e4037817c5d176911bdea>:0 
  at OpenTK.GameWindow.Run (System.Double updates_per_second, System.Double frames_per_second) [0x0008f] in <6764f93f9a3e4037817c5d176911bdea>:0 
  at OpenTK.GameWindow.Run () [0x00000] in <6764f93f9a3e4037817c5d176911bdea>:0 
  at OpenBve.MainLoop.StartLoopEx (OpenBve.formMain+MainDialogResult result) [0x001b0] in <eb4ae41797e5478ba3982f0076e9d8ed>:0 
  at OpenBve.Program.Main (System.String[] args) [0x004c0] in <eb4ae41797e5478ba3982f0076e9d8ed>:0   at System.Environment.get_StackTrace () [0x00000] in <8f2c484307284b51944a1a13a14c0266>:0 
  at OpenBve.Program.Main (System.String[] args) [0x00000] in <eb4ae41797e5478ba3982f0076e9d8ed>:0 
Current program memory usage: 1139.02mb
System memory free: 0mb
```

</details>

**Comment by leezer3** [comment]
Thanks, that's a very odd log.

Stack trace suggests it dies somewhere in this function when attempting to update the train exterior / cab:
https://github.com/leezer3/OpenBVE/blob/release/source/OpenBVE/Simulation/TrainManager/Car/Car.cs#L592

Not where I was expecting it to die at all, and I haven't got any immediate ideas.
Did you upgrade a working version of 1.6.0- I'm wondering if a remnant is hanging around somewhere?

Will try and do some testing over the next few days and see if I can reproduce.

**Comment by leezer3** [comment]
@s520 @ginga81 Can either of you reproduce this at all on a 'real' Debian machine?

Will try a completely fresh VM on Monday, and could probably even roll a real one.

This feels somewhat related to the camera properties changes in 1.7, but don't know...
Might also be masking a missing Mono library- Are we pulling something in that needs to be in the deb dependancies section now?

**Comment by s520** [comment]
I don't have a real Debian machine.
@ ginga81 has a real Ubuntu machine.
However, a huge typhoon is currently hitting Japan.
As a result, he takes some time to validate...

**Comment by leezer3** [comment]
Of course, I forgot about that :(
I hope that both of you stay safe, and well throughout this.

---

Removing mono-complete from my test VM and just pulling in the package dependencies seems to crash in the way described, so that's good!
Just need to figure out what the missing dependency is, which may be a real PITA....

@Symbian9 - Installing mono-complete will *probably* get you going until we get this sorted out.

**Comment by s520** [comment]
Thank you for your concern.

If the problem is solved by installing mono-complete, there was something similar to this problem before.
https://github.com/leezer3/OpenBVE/issues/369

There may be a relationship...

**Comment by s520** [comment]
@leezer3 
I'm safe, please don't worry.
Now, this problem seems to have been solved, but do I need to validate it?

**Comment by leezer3** [comment]
As I've got this reproduced on a VM I don't think so :)
Was just concerned it was something odd which only showed up on a physical box.

I've pushed 1.7.0.3, which solves both cases I'm aware of (this and http://bveworldwide.forumotion.com/t1791-using-a-controller-with-openbve#18935 ) and will be updating the website shortly.

**N.B.** I also bumped the minimum Mono version for this one to 5.20 as previously discussed, although that's not strictly required yet, but will be when the first release with the new renderer goes out. 
Will see if we get any complaints on that score before doing anything else.....
